synapse/cortex.py

```diff
@@ -916,6 +916,15 @@ async def getPermDefs(self):
         '''
         return await self.cell.getPermDefs()
 
+    async def getAxonUpload(self):
+        self.user.confirm(('axon', 'upload'))
+        return await s_axon.UpLoadShare.anit(self.cell.axon, self.link)
```

Comment:
This doesn't seem to work correctly for a remote axon:
```storm> !pushfile setup.cfg
uploading file: setup.cfg
Traceback (most recent call last):
  File "/home/cisphyx/git/synapse/synapse/lib/cli.py", line 431, in runCmdLine
    ret = await cmdo.runCmdLine(line)
  File "/home/cisphyx/git/synapse/synapse/lib/cli.py", line 54, in runCmdLine
    return await self.runCmdOpts(opts)
  File "/home/cisphyx/git/synapse/synapse/tools/storm.py", line 126, in runCmdOpts
    size, sha256 = await upload.save()
  File "/home/cisphyx/git/synapse/synapse/telepath.py", line 350, in __call__
    return await self.proxy.task(todo, name=self.share)
  File "/home/cisphyx/git/synapse/synapse/telepath.py", line 753, in task
    return await self.taskv2(todo, name=name)
  File "/home/cisphyx/git/synapse/synapse/telepath.py", line 706, in taskv2
    return s_common.result(retn)
  File "/home/cisphyx/git/synapse/synapse/common.py", line 776, in result
    raise ctor(**info)
synapse.exc.NotMsgpackSafe: NotMsgpackSafe: efile='msgpack.py' eline=45 ename='en' esrc='raise s_exc.NotMsgpackSafe(mesg=mesg) from e' mesg="can not serialize 'generator' object: ('t2:init', {'todo':"

error: NotMsgpackSafe: efile='msgpack.py' eline=45 ename='en' esrc='raise s_exc.NotMsgpackSafe(mesg=mesg) from e' mesg="can not serialize 'generator' object: ('t2:init', {'todo':"
```

Reply:
From the cortex:
```021-08-25 10:31:43,768 [ERROR] error during task: save [daemon.py:t2call:MainThread:MainProcess]
Traceback (most recent call last):
  File "/home/cisphyx/git/synapse/synapse/lib/msgpack.py", line 41, in en
    return pakr.pack(item)
  File "msgpack/_packer.pyx", line 292, in msgpack._cmsgpack.Packer.pack
  File "msgpack/_packer.pyx", line 298, in msgpack._cmsgpack.Packer.pack
  File "msgpack/_packer.pyx", line 295, in msgpack._cmsgpack.Packer.pack
  File "msgpack/_packer.pyx", line 264, in msgpack._cmsgpack.Packer._pack
  File "msgpack/_packer.pyx", line 231, in msgpack._cmsgpack.Packer._pack
  File "msgpack/_packer.pyx", line 264, in msgpack._cmsgpack.Packer._pack
  File "msgpack/_packer.pyx", line 264, in msgpack._cmsgpack.Packer._pack
  File "msgpack/_packer.pyx", line 289, in msgpack._cmsgpack.Packer._pack
TypeError: can not serialize 'generator' object

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/cisphyx/git/synapse/synapse/daemon.py", line 131, in t2call
    valu = await valu
  File "/home/cisphyx/git/synapse/synapse/axon.py", line 229, in save
    await self.axon.save(sha256, genr())
  File "/home/cisphyx/git/synapse/synapse/telepath.py", line 350, in __call__
    return await self.proxy.task(todo, name=self.share)
  File "/home/cisphyx/git/synapse/synapse/telepath.py", line 753, in task
    return await self.taskv2(todo, name=name)
  File "/home/cisphyx/git/synapse/synapse/telepath.py", line 697, in taskv2
    await link.tx(mesg)
  File "/home/cisphyx/git/synapse/synapse/lib/link.py", line 221, in tx
    byts = s_msgpack.en(mesg)
  File "/home/cisphyx/git/synapse/synapse/lib/msgpack.py", line 45, in en
    raise s_exc.NotMsgpackSafe(mesg=mesg) from e
synapse.exc.NotMsgpackSafe: NotMsgpackSafe: mesg="can not serialize 'generator' object: ('t2:init', {'todo':"
```

Reply:
Great catch, thanks!